src/Catel.Core/Catel.Core.Shared/IoC/TypeRequestPath.cs

```diff
@@ -22,68 +20,88 @@ public class TypeRequestPath : ITypeRequestPath
         /// </summary>
         private static readonly ILog Log = LogManager.GetCurrentClassLogger();
 
-        private readonly List<TypeRequestInfo> _typePath = new List<TypeRequestInfo>();
+        private readonly TypeRequestInfo[] _typePath;
         
-        private bool _isValid = true;
+        private TypeRequestPath(TypeRequestInfo[] typePath, string name)
+        {
+            _typePath = typePath;
+            Name = name;
+        }
+
+        private TypeRequestPath(TypeRequestInfo[] typePath, string name)
+        {
+            _typePath = typePath;
+            Name = name;
+        }
 
         /// <summary>
-        /// Initializes a new instance of the <see cref="TypeRequestPath" /> class.
+        /// Creates root of type request path
         /// </summary>
-        /// <param name="typeRequestInfo">The type request info.</param>
-        /// <param name="ignoreValueTypes">If set to <c>true</c>, this type path will ignore value types.</param>
-        /// <param name="name">The name, can be <c>null</c>.</param>
-        /// <exception cref="ArgumentNullException">The <paramref name="typeRequestInfo" /> is <c>null</c>.</exception>
-        public TypeRequestPath(TypeRequestInfo typeRequestInfo, bool ignoreValueTypes = true, string name = null)
-            : this(new [] { typeRequestInfo }, ignoreValueTypes, name) { }
+        /// <param name="name">Path's name</param>
+        /// <returns></returns>
+        public static TypeRequestPath Root(string name = null)
+        {
+            return new TypeRequestPath(new TypeRequestInfo[0], name);
```

Comment:
Can we create a static instance of new TypeRequestInfo[0] to prevent additional allocations?